internal/fuzzbinder/fuzzscraper.go

```diff
@@ -0,0 +1,335 @@
+// summary contains fine-grained data related to coverage metrics.
+// CoverageSummary contains the coverage report summary.
+type CoverageSummary struct {
+	Data    []CoverageSummaryData `json:"data"`
+	Type    interface{}           `json:"type"`
+	Version interface{}           `json:"version"`
+}
+
+// CoverageSummaryData contains the data of the coverage report summary.
+type CoverageSummaryData struct {
+	Files  interface{}                     `json:"files"`
+	Totals map[string](map[string]float64) `json:"totals"`
+}
+
+// getBucket gets a Google Cloud Storage bucket given its name.
+func getBucket(bucketName string) (*storage.BucketHandle, error) {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	bucket := client.Bucket(bucketName)
+	return bucket, nil
+}
+
+// listBlobs gets all the objects in a Google Cloud Storage bucket
+// under a given relative path.
+func listBlobs(bucket *storage.BucketHandle, relativePath string) ([]string, error) {
+	var blobs []string
+	ctx := context.Background()
+	query := &storage.Query{Prefix: relativePath}
+	objects := bucket.Objects(ctx, query)
+	for {
+		attrs, err := objects.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Bucket.Objects: %v", err)
+		}
+		blobs = append(blobs, attrs.Name)
+	}
+	return blobs, nil
+}
+
+// getBlob gets the file reader of a blob in a Google Cloud Storage bucket.
+func getBlob(bucket *storage.BucketHandle, blobName string) (*storage.Reader, error) {
+	ctx := context.Background()
+	reader, err := bucket.Object(blobName).NewReader(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Object(%q).NewReader: %v", blobName, err)
+	}
+	defer reader.Close()
+	return reader, nil
+}
+
+// getRevisionFromFile gets the revision hash of the source code for which OSS-Fuzz coverage
+// reports were generated on a given day, given the file where the revision hash is saved.
+func getRevisionFromFile(content []byte, projectName string) (string, error) {
+	var payload map[string](map[string]string)
+	err := json.Unmarshal(content, &payload)
+	if err != nil {
+		return "", fmt.Errorf("unmarshal: %v", err)
+	}
+	// Get the revisionHash using the file structure defined by OSS-Fuzz.
+	revisionHash := payload[fmt.Sprintf("/src/%s", projectName)]["rev"]
+	return revisionHash, nil
+}
+
+// parseCoverageSummary gets the branch coverage and line coverage from a coverage report summary.
+func parseCoverageSummary(content []byte) (map[string]float64, map[string]float64, error) {
+	var summary CoverageSummary
+	err := json.Unmarshal(content, &summary)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unmarshal: %v", err)
+	}
+	// Return branch coverage and line coverage using the coverage summary structure.
+	return summary.Data[0].Totals["branches"], summary.Data[0].Totals["lines"], nil
```

Comment:
Why not return the `CoverageSummary` object and let the caller choose what data to use? 
Or perhaps define a custom type instead of returning two maps. 
Feel free to leave it for a future PR.